lib/SharedLib/CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+# We supply empty.cpp because CMake does not allow creating a library without sources.
+add_library(TTMLIRSO SHARED empty.cpp)
+
+set(METAL_LIB_DIR "${CMAKE_SOURCE_DIR}/third_party/tt-metal/src/tt-metal-build/lib")
+
+add_library(ttnn SHARED IMPORTED)
+set_property(TARGET ttnn PROPERTY IMPORTED_LOCATION "${METAL_LIB_DIR}/_ttnn.so")
+
+add_dependencies(TTMLIRSO
+    TTNNTargetFlatbuffer
+    MLIRTTDialect
+    MLIRTTIRDialect
+    MLIRTTNNDialect
+    MLIRTTKernelDialect
+    TTMLIRTTIRToTTNN
+    MLIRTTMetalDialect
+    MLIRTTIRTransforms
+    MLIRTTNNTransforms
+    MLIRTTIRAnalysis
+    MLIRTTNNPipelines
+    TTMLIRTTNNToEmitC
+    TTRuntime
+    TTRuntimeTTNN)
+
+target_link_libraries(TTMLIRSO PRIVATE
+    # LLVM
+    LLVM
+    MLIR
+
+    # Forces the inclusion of all symbols in the shared object
+    # This is necessary because the JIT will not be able to find the symbols otherwise
+    -Wl,--whole-archive
+    # MLIR libraries
+    TTNNTargetFlatbuffer
+    MLIRTTDialect
+    MLIRTTIRDialect
+    MLIRTTNNDialect
+    MLIRTTKernelDialect
+    TTMLIRTTIRToTTNN
+    MLIRTTMetalDialect
+    MLIRTTIRTransforms
+    MLIRTTNNTransforms
+    MLIRTTIRAnalysis
+    MLIRTTNNPipelines
+    TTMLIRTTNNToEmitC
+    TTRuntime
+    TTRuntimeTTNN
+    # MLIR libraries
+    -Wl,--no-whole-archive
```

Comment:
Instead of passing these linker flags we should change the `target_link_libraries`, `PRIVATE` to `PUBLIC`, which should export the symbols of these static libraries.

Reply:
I don't think this work because `tt-mlir` is built in separate project and its targets are not visible to `tt-forge`, hence adding public wouldn't change anything.

Reply:
I think `PUBLIC` implies `-Wl,--no-whole-archive`, would you mind trying?

Reply:
Ok I don't want to block the review because we should get this in.  But something else must be messed up somewhere in the cmake tree, this is what `PUBLIC` means so I'm not sure why it would be hiding the symbols.